File: programmers/lv0/코딩기초트레이닝/조건문/코드처리하기/Solution.js

const solution = (code) => {
    let answer = ''
    let mode = 0
    for (let i = 0; i < code.length; i++) {
        if (code[i] === '1') {
            mode = (mode + 1) % 2
        } else {
            if (i % 2 === mode) answer += code[i]
        }
    }
    return answer.length === 0 ? 'EMPTY' : answer
}

console.log(solution("abc1abc1abc"));

File: programmers/lv0/코딩기초트레이닝/조건문/코드처리하기/Solution.swift
import Foundation

func solution(_ code:String) -> String {
    var answer = ""
    var mode = 0

    code.enumerated().forEach {
        if $0.element == "1" {
            mode ^= 1
        }
        else {
            if ($0.offset % 2 == mode) {
                answer += String($0.element)
            }
        }
    }
    return answer.isEmpty ? "EMPTY" : answer
}

print(solution("abc1abc1abc"))
